File: backend/goat_listing/src/main/java/com/goat/identity/domain/valueobjects/Email.java
package com.goat.identity.domain.valueobjects;

import java.util.Objects;
import java.util.regex.Pattern;

/**
 * Value Object que representa un email válido.
 * Encapsula la validación de formato de email.
 */
public final class Email {
    private static final Pattern EMAIL_PATTERN = Pattern.compile(
        "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$"
    );

    private final String value;

    private Email(String value) {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("El email no puede estar vacío");
        }
        String trimmedValue = value.trim().toLowerCase();
        if (!EMAIL_PATTERN.matcher(trimmedValue).matches()) {
            throw new IllegalArgumentException("El formato del email no es válido");
        }
        this.value = trimmedValue;
    }

    public static Email of(String value) {
        return new Email(value);
    }

    public String getValue() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Email email = (Email) o;
        return Objects.equals(value, email.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value);
    }

    @Override
    public String toString() {
        return value;
    }
}

File: backend/goat_catalog/src/goat_catalog/notifications/domain/value_objects/email.py
"""Value object para Email."""

from dataclasses import dataclass
import re


@dataclass(frozen=True)
class Email:
    """Value object que representa un email válido."""

    value: str

    def __post_init__(self) -> None:
        """Valida que el email tenga un formato válido."""
        if not self.value or not isinstance(self.value, str):
            raise ValueError("El email no puede estar vacío")
        
        email_pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
        if not re.match(email_pattern, self.value):
            raise ValueError(f"El email '{self.value}' no tiene un formato válido")
